service/iot-service-client/src/main/java/com/microsoft/azure/sdk/iot/service/transport/amqps/AmqpConnectionHandler.java

```diff
@@ -76,12 +94,42 @@ protected AmqpConnectionHandler(String hostName, String userName, String sasToke
         this.sslContext = sslContext; // if null, a default SSLContext will be generated for the user
     }
 
+    protected AmqpConnectionHandler(
+            String hostName,
+            TokenCredential authenticationTokenProvider,
+            TokenCredentialType authorizationType,
+            IotHubServiceClientProtocol iotHubServiceClientProtocol,
+            ProxyOptions proxyOptions,
+            SSLContext sslContext)
+    {
+        if (Tools.isNullOrEmpty(hostName))
+        {
+            throw new IllegalArgumentException("hostName can not be null or empty");
+        }
+
+        if (iotHubServiceClientProtocol == null)
+        {
+            throw new IllegalArgumentException("iotHubServiceClientProtocol cannot be null");
+        }
+
```

Comment:
null check for authenticationTokenProvider and authorizationType.

Reply:
I'm going to do those null checks at a higher level in the code so that we don't need null checks in every class